cli/util/fs.rs

```diff
@@ -300,14 +348,17 @@ impl<TFilter: Fn(&Path, &FilePatterns) -> bool> FileCollector<TFilter> {
         let file_type = e.file_type();
         let is_dir = file_type.is_dir();
         let c = e.path().to_path_buf();
-        if file_patterns.exclude.matches_path(&c)
-          || !is_dir
-            && !file_patterns
-              .include
-              .as_ref()
-              .map(|i| i.matches_path(&c))
-              .unwrap_or(true)
-        {
+        let maybe_gitignore =
+          maybe_git_ignores.as_mut().and_then(|git_ignores| {
+            let dir_path = if is_dir { &c } else { c.parent()? };
```

Comment:
Add context to `c.parent()`?

Reply:
Do you mean anyhow context? Can't for this because it's an option and this should never happen because `c.parent()` would only be undefined for the root directory, in which case it would match on `is_dir`. I just did `?` to not unwrap.

Reply:
Yeah I meant anyhow context. Disregard if it's an option.